sn_interface/src/network_knowledge/section_authority_provider.rs

```diff
@@ -325,5 +350,18 @@ pub mod test_utils {
 
             (signed_sap, nodes, secret_key_set.secret_key())
         }
+
+        pub fn _gen_sk_set<R: RngCore>(
```

Comment:
The `_` here is throwing me off, It seems like you want to use this helper in tests so it's not intended to be private. 

How about just `gen_sk_set`
```suggestion
        pub fn gen_sk_set<R: RngCore>(
```